actionpack/lib/action_dispatch/railtie.rb

```diff
@@ -52,5 +52,15 @@ class Railtie < Rails::Railtie # :nodoc:
 
       ActionDispatch.test_app = app
     end
+
+    initializer "action_dispatch.set_test_utils" do
+      ActiveSupport.on_load(:action_dispatch_integration_test) do
+        require "active_job"
+        require "action_mailer"
+
+        include ActiveJob::TestHelper
```

Comment:
This need to be the opposite. It should be the Active Job and Action Mailer railtie that includes those modules in the `action_dispatch_integration_test` for the case were your app don't use one of those frameworks.

Reply:
That makes a lot of sense. I was struggling about loading both gems in the ActionDispatch railtie. I updated this patch according to your suggestions. Thanks!